src/activities/quizzes/attempts/QuizAttemptsEntity.js

```diff
@@ -83,43 +83,85 @@ export class QuizAttemptsEntity extends Entity {
 		return attemptsAllowed !== this.attemptsAllowed();
 	}
 
+	/**
+	 * Returns an update attempts action if one exists
+	 * @param {number} attemptsAllowed number of attempts allowed
+	 */
+
+	_generateAttemptsAllowedAction(attemptsAllowed) {
+		let action;
+		if (!this.canUpdateAttemptsAllowed()) {
+			return;
+		}
+
+		if (this._entity) {
+			action = this._entity.getActionByName(Actions.quizzes.attempts.updateAttemptsAllowed);
+		}
+
+		if (!action) {
+			return;
+		}
+
+		const fields = [
+			{ name: 'attemptsAllowed', value: attemptsAllowed },
+		];
+
+		return { action, fields };
+
+	}
+
 	async setAttemptsAllowed(attemptsAllowed) {
 		if (!attemptsAllowed || !this._hasAttemptsAllowedChanged(attemptsAllowed)) return;
 		if (!this.canUpdateAttemptsAllowed()) {
 			return;
 		}
-		const {action, fields } = this._generateAttemptsAllowedAction(attemptsAllowed);
+		const {action, fields} = this._generateAttemptsAllowedAction(attemptsAllowed);
 		if (!action) {
 			return;
 		}
 		await performSirenAction(this._token, action, fields);
 	}
 
+	_hasOverallGradeCalculationTypeChanged(calculationType) {
+		return calculationType !== this.overallGradeCalculationType();
+	}
+
 	/**
-	 * Returns an update attempts action if one exists
-	 * @param {number} attemptsAllowed number of attempts allowed
+	 * Returns an update overall grade calculation type action if one exists
+	 * @param {string} calculationType number of attempts allowed
 	 */
 
-	_generateAttemptsAllowedAction(attemptsAllowed) {
+	_generateOverallGradeCalculationTypeAction(calculationType) {
 		let action;
-		if (!this.canUpdateAttemptsAllowed()) {
+		if (!this.canUpdatecalculationType()) {
 			return;
 		}
 
 		if (this._entity) {
-			action = this._entity.getActionByName(Actions.quizzes.attempts.updateAttemptsAllowed);
+			action = this._entity.getActionByName(Actions.quizzes.attempts.updatecalculationType);
 		}
 
 		if (!action) {
 			return;
 		}
 
 		const fields = [
-			{ name: 'attemptsAllowed', value: attemptsAllowed },
+			{ name: 'overallGradeCalculationType', value: calculationType },
 		];
 
 		return { action, fields };
 
 	}
 
+	async setOverallGradeCalculationType(calculationType) {
+		if (!calculationType || !this._hasOverallGradeCalculationTypeChanged(calculationType)) return;
+		if (!this.canUpdateOverallGradeCalculation()) {
+			return;
+		}
+		const {action, fields} = this._generateOverallGradeCalculationTypeAction(calculationType);

```

Comment:
Needs to be
`const {action, fields} = this._generateOverallGradeCalculationTypeAction(calculationType) || {};`
Or it is a `TypeError` as it can't destructure `action` and `fields`